libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -589,29 +601,49 @@ - (BOOL)hasCommunityChanged {
     }
 }
 
-- (void)userChanged:(SFUserAccountChange)change {
-    if (![self.lastChangedOrgId isEqualToString:self.currentUser.credentials.organizationId]) {
-        self.lastChangedOrgId = self.currentUser.credentials.organizationId;
+- (void)userChanged:(SFUserAccount *)user change:(SFUserAccountChange)change {
+    [self notifyUserChange:SFUserAccountManagerDidChangeUserDataNotification withUser:user andChange:change];
+}
+
+- (void)notifyCurrentUserChange:(SFUserAccountChange)change {
```

Comment:
Also, under some circumstances, consumers may not actually care that the current user has switched.  Largely that feels like an app-oriented thing.